gengokit/gengokit.go

```diff
@@ -329,14 +356,14 @@ func applyTemplate(templBytes []byte, templName string, executor interface{}) (*
 // formatCode takes a string representing golang code and attempts to return a
 // formated copy of that code.  If formatting fails, a warning is logged and
 // the original code is returned.
-func formatCode(code []byte) []byte {
+func formatCode(code []byte) ([]byte, error) {
 	formatted, err := format.Source(code)
 
 	if err != nil {
 		log.WithError(err).Warn("Code formatting error, generated service will not build, outputting unformatted code")
-		// Set formatted to code so at least we get something to examine
-		formatted = code
+		// return code so at least we get something to examine
+		return code, err
```

Comment:
This is kind of unidiomatic. Typcially if a function returns an error, it returns the zero value of any other return values. The caller typically checks for an error and acts on that. The caller in this case could know that this function returns other values even when an error is returns, and inspect that, but that's a one-off that requires knowledge about implementation details of the function that is being called, which is pretty un-Go-like in that it's a sort of "magic."